### PR TITLE
Use the scope matcher if an element renders differently in BE and FE

### LINKED ISSUE #1187: HTML-Inhaltselemente im RSS-Feed
**Affected version**
Contao 4.8.7

**Description**
HTML-Inhaltslemente von Nachrichten werden im RSS-Feed als Text ausgegeben. Aus `<div class="panel">` wird `<pre>&lt;div class=&quot;panel&quot;&gt;</pre>`
Beispiel: https://km-kd.de/share/news.xml

Erwartetes Ergebnis wäre, dass das HTML im RSS-Feed als solches ausgegeben wird.

### PR DESCRIPTION
**Pull request by leofeyer**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #1187
| Docs PR or issue | -

Shall I also try to fix https://github.com/contao/core/issues/7587 here?

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/Controller.php [hunk 1]
```diff
@@ -1132,6 +1133,23 @@ public static function redirect($strLocation, $intStatus=303)
 		throw new RedirectResponseException($strLocation, $intStatus);
 	}
 
+	/**
+	 * Get the current request from the request stack
+	 *
+	 * @return Request
+	 */
+	public static function getCurrentRequest()
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/Controller.php]: Not sure if you like the new method here. I wanted to save myself from copying the exception multiple times.

File: core-bundle/src/Resources/contao/elements/ContentCode.php [hunk 2]
```diff
@@ -30,7 +30,9 @@ class ContentCode extends ContentElement
 	 */
 	public function generate()
 	{
-		if (TL_MODE == 'BE')
+		$request = Controller::getCurrentRequest();
+
+		if (System::getContainer()->get('contao.routing.scope_matcher')->isBackendRequest($request))
```
Comment (ausi) [inline core-bundle/src/Resources/contao/elements/ContentCode.php]:
```suggestion
		$request = System::getContainer()->get('request_stack')->getCurrentRequest();

		if (null !== $request && System::getContainer()->get('contao.routing.scope_matcher')->isBackendRequest($request))
```

I think we should not throw an exception in the content elements if there is no request. If I want to render an element in an command there might be no request.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/elements/ContentCode.php]: But how do we know whether to apply `htmlspecialchars()` then?

Comment (ausi) [reply core-bundle/src/Resources/contao/elements/ContentCode.php]: Default for content element rendering ist frontend context IMO.
If there is a request and it is in backend scope we should render it for the backend scope (applying `htmlspecialchars()` in this case).

Comment (leofeyer) [reply core-bundle/src/Resources/contao/elements/ContentCode.php]: Agreed. I will check whether all elements follow this logic.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/elements/ContentCode.php]: Adjusted in cb3e80f61d17cff20b85898922c3ba55bf2789af.

**Comment by ausi** [comment]
Did you skip several content elements on purpose? like `ContentAccordionStart` and `ContentAccordionStop`?

**Comment by leofeyer** [comment]
I have left all the `if (TL_MODE == 'BE') { renderBackEndWildcardTemplate(); }` snippets as is. Do you think it would be better to adjust them as well?

**Comment by ausi** [comment]
> Do you think it would be better to adjust them as well?

I guess so, because in the RSS feed (and in newsletters) the backend wildcard output ist most probably not desired.

**Comment by leofeyer** [comment]
See 947d77ab2af7dee6186e3ee5456c8d5eb591202d.

**Review by ausi — APPROVED** [review]

**Review by bytehead — APPROVED** [review]

**Comment by aschempp** [comment]
I think instead of modifying the current request, we should have pushed a new request to the stack. Shall I prepare a PR for that?

**Comment by leofeyer** [comment]
No, I don't think that is necessary. Temporarily adjusting the current request works fine.

**Pull request merged**